app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java

```diff
@@ -28,20 +37,51 @@
  * TODO Rename to remove new part.
  */
 public class NewProfileActivity extends AppCompatActivity {
+    public final static String AUTH_INTENT = "isAuth";
+    public final static String OTHER_INTENT = "otherId";
+
+    private boolean isAuthProfile;
+
+    private OtherAccount otherAccount = null;
+
 
     private View selectedCollected = null;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_new_profile);
 
-        StatusBarHandler.StatusBarTransparent(this);
+        // Get the starting intent
+        Intent startingIntent = getIntent();
+        isAuthProfile = startingIntent.getBooleanExtra(AUTH_INTENT, false);
+
+        if(!isAuthProfile) {
+            String otherId = startingIntent.getStringExtra(OTHER_INTENT);
+            new Thread(() -> {
+                otherAccount = OtherAccount.getInstance(otherId);
+                runOnUiThread(() -> {
+                    setContentView(R.layout.activity_new_profile);
+
+                    StatusBarHandler.StatusBarTransparent(this);
```

Comment:
These two are in both if and else. Is it needed because of the  `runOnUiThread()` or could they be preformed before the if condition?

Reply:
Yep. As in the case of another user we need to load its data first, I prefer not to show the layout before the data has been loaded.
What do you think?